homeassistant/components/radarr/__init__.py

```diff
@@ -1 +1,68 @@
-"""The radarr component."""
+"""The Radarr component."""
+from __future__ import annotations
+
+from aiopyarr.models.host_configuration import PyArrHostConfiguration
+from aiopyarr.radarr_client import RadarrClient
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_API_KEY, CONF_URL, CONF_VERIFY_SSL, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.device_registry import DeviceEntryType
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DEFAULT_NAME, DOMAIN
+from .coordinator import RadarrDataUpdateCoordinator
+
+PLATFORMS = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Radarr from a config entry."""
+    host_configuration = PyArrHostConfiguration(
+        api_token=entry.data[CONF_API_KEY],
+        verify_ssl=entry.data[CONF_VERIFY_SSL],
+        url=entry.data[CONF_URL],
+    )
+    radarr = RadarrClient(
+        host_configuration=host_configuration,
+        session=async_get_clientsession(hass, entry.data[CONF_VERIFY_SSL]),
+    )
+    coordinator = RadarrDataUpdateCoordinator(hass, host_configuration, radarr)
+    await coordinator.async_config_entry_first_refresh()
+    entry.async_on_unload(entry.add_update_listener(_async_update_listener))
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = coordinator
+    hass.config_entries.async_setup_platforms(entry, PLATFORMS)
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    if unload_ok := await hass.config_entries.async_unload_platforms(entry, PLATFORMS):
+        hass.data[DOMAIN].pop(entry.entry_id)
+    return unload_ok
+
+
+async def _async_update_listener(hass: HomeAssistant, entry: ConfigEntry) -> None:
+    """Handle options update."""
+    await hass.config_entries.async_reload(entry.entry_id)
+
+
+class RadarrEntity(CoordinatorEntity):
+    """Defines a base Radarr entity."""
+
+    coordinator: RadarrDataUpdateCoordinator
+
+    def __init__(self, coordinator: RadarrDataUpdateCoordinator) -> None:
+        """Initialize the Radarr entity."""
+        super().__init__(coordinator)
+        self._attr_device_info = DeviceInfo(
+            configuration_url=coordinator.host_configuration.url,
+            entry_type=DeviceEntryType.SERVICE,
+            identifiers={(DOMAIN, coordinator.config_entry.entry_id)},
+            manufacturer=DEFAULT_NAME,
+            name=DEFAULT_NAME,
```

Comment:
Since it's 1 config entry -> 1 service, you could keep it in sync with the config entry title. 
```suggestion
            name=coordinator.config_entry.title,
```